nixos/modules/services/web-apps/nextcloud.nix

```
@@ -309,7 +309,7 @@ in {
       '';
       example = literalExpression ''
         {
-          inherit (pkgs.nextcloud25Packages.apps) mail calendar contact;
+          inherit (pkgs.nextcloud31Packages.apps) mail calendar contact;
```

We could in theory do
```suggestion
          inherit (config.services.nextcloud.package.packages.apps) mail calendar contact;
```

But it looks even more dumb

---

but it is more correct

---

That doesn't exist

---

oh, `passthru.packages`. Even longer.

---

Uhm did something change around that on unstable that I'm missing?

I do have

```nix
    extraApps = with config.services.nextcloud.package.packages.apps; {
      inherit
        bookmarks
        calendar
        contacts
        cospend
        #maps
        notes
        phonetrack
        previewgenerator
        user_oidc
        ;
    };
```

in my config.

Everything in `passthru` should be available as attri from the drv itself fwiw.

That being said, I strongly prefer that suggestion as well given that we shouldn't have to update yet another thing per update.

---

Sorry, forgot this one. I do think that we are making an overly long attribute which doesn't really help. Maybe we should make an extra option to choose bundled apps.

> Everything in `passthru` should be available as attri from the drv itself fwiw.

I guess that's an restriction turned on in ofborg? I did see some people adding passthru in some places, but didn't investigate.

---

It might be because I overlay the nextcloud package and don't passthru passthru.